pkg/fswatcher/fswatcher.go

```diff
@@ -4,41 +4,73 @@
 package fswatcher
 
 import (
-	"errors"
-	"fmt"
+	"bytes"
+	"crypto/sha256"
+	"io"
 	"os"
-	"path/filepath"
+	"sync"
+	"testing"
 
-	"github.com/fsnotify/fsnotify"
-	"github.com/sirupsen/logrus"
+	"github.com/cilium/cilium/pkg/time"
+)
+
+const (
+	// how often tracked targets are checked for changes by default
+	defaultInterval = 5 * time.Second
+
+	// when fswatcher detects that it runs in a test, it will refresh things much faster
+	testInterval = 50 * time.Millisecond
```

Comment:
Do we want to backport this PR to older versions of Cilium? If not, consider using [https://go.dev/blog/synctest instead](https://go.dev/blog/synctest#testing-time)

Reply:
It looks very tempting to try, but I'm a bit hesitant to use an experimental feature that's still behind a feature flag which would require a change in the build requirements. Maybe better to let it bake for another version :)

Reply:
Ah, my bad - I thought we already did use synctest in Cilium and didn't realize it was still behind a feature flag.